lib/omnibus/git_cache.rb

```diff
@@ -22,6 +22,18 @@ class GitCache
     include Util
     include Logging
 
+    # The serial number represents compatibility of a cache entry with the
+    # current version of the omnibus code base. Any time a change is made to
+    # omnibus that makes the code incompatible with any cache entries created
+    # before the code change, the serial number should be incremented.
+    #
+    # For example, if a code change generates content in the `install_dir`
+    # before cache snapshots are taken, any snapshots created before upgrade
+    # will not have the generated content, so these snapshots would be
+    # incompatible with the current omnibus codebase. Incrementing the serial
+    # number ensures these old shapshots will not be used in subsequent builds.
+    SERIAL_NUMBER = 1
```

Comment:
I'll elaborate.

If you use `major.minor`, then:
* You can have false negatives, e.g., you can do a bug fix that looks fine but is actually incompatible with existing caches, and builds break.
* You will have false positives, e.g., a new feature (like support for a new platform) is added that doesn't affect cache validity at all, but resets your cache anyway.

If you use the SHA of the omnibus code then:
* You get a lot of false positives.
* You never get a false negative

Given the problems that clean caches can cause for us, we want to make the number of false positives as low as possible, so that makes the SHA-based option less attractive.